modules/core.go

```diff
@@ -23,133 +23,145 @@
+// TryGet returns true if a module was stored in ptr, false otherwise.
+//
+// NOTE: ptr must be a non-nil pointer to a type that has been provided to the module system.
+//
+// Example:
+//
+//	builder := modules.New()
+//	builder.Provide(MyModuleImpl{}, new(MyModule))
+//	mods = builder.Build()
+//
+//	var module MyModule
+//	if mods.TryGet(&module) {
+//		// success
+//	}
+func (mods Core) TryGet(ptr any) bool {
+	v := reflect.ValueOf(ptr)
+	if !v.IsValid() {
+		panic("nil value given")
+	}
+	pt := v.Type()
+	if pt.Kind() != reflect.Ptr {
+		panic("only pointer values allowed")
+	}
 
-// MetricsLogger returns the metrics logger.
-func (mods Core) MetricsLogger() MetricsLogger {
-	if mods.metricsLogger == nil {
-		return NopLogger()
+	for _, m := range mods.modules {
+		mv := reflect.ValueOf(m)
+		if mv.Type().AssignableTo(pt.Elem()) {
+			v.Elem().Set(mv)
+			return true
+		}
 	}
-	return mods.metricsLogger
-}
 
-// EventLoop returns the event loop.
-func (mods Core) EventLoop() *eventloop.EventLoop {
-	return mods.eventLoop
+	return false
 }
 
-// MetricsEventLoop returns the metrics event loop.
-// The metrics event loop is used for processing of measurement data.
+// Get finds a module for ptr.
 //
-// Deprecated: The metrics event loop is no longer separate from the main event loop. Use EventLoop() instead.
-func (mods Core) MetricsEventLoop() *eventloop.EventLoop {
-	return mods.EventLoop()
+// NOTE: ptr must be a non-nil pointer to a type that has been provided to the module system.
+// Get panics if ptr is not a pointer, or if a compatible module is not found.
+//
+// Example:
+//
+//	builder := modules.New()
+//	builder.Provide(MyModuleImpl{}, new(MyModule))
+//	mods = builder.Build()
+//
+//	var module MyModule
+//	mods.Get(&module)
+func (mods *Core) Get(ptr any) {
+	if !mods.TryGet(ptr) {
+		panic(fmt.Sprintf("module of type %s not found", reflect.TypeOf(ptr).Elem()))
+	}
 }
 
-// GetModuleByType makes it possible to get a module based on its real type.
-// This is useful for getting modules that do not implement any known module interface.
-// The method returns true if a module was found, false otherwise.
-//
-// NOTE: dest MUST be a pointer to a variable of the desired type.
-// For example:
-//  var module MyModule
-//  if mods.GetModuleByType(&module) { ... }
-func (mods Core) GetModuleByType(dest any) bool {
-	outType := reflect.TypeOf(dest)
-	if outType.Kind() != reflect.Ptr {
-		panic("invalid argument: out must be a non-nil pointer to an interface variable")
-	}
-	targetType := outType.Elem()
-	if m, ok := mods.modulesByType[targetType]; ok {
-		reflect.ValueOf(dest).Elem().Set(reflect.ValueOf(m))
-		return true
+// GetAll finds a module for all the given pointers.
+//
+// NOTE: pointers must only contain non-nil pointers to types that have been provided to the module system.
+// GetAll panics if one of the given pointers is not a pointer, or if a compatible module is not found.
+func (mods *Core) GetAll(pointers ...any) {
```

Comment:
Suggest the remove `GetAll` and migrate the `...any` to the `Get` method, so that you can call `Get` with one or many arguments. Zero arguments is a no-op so there is no need to panic, but it is a programmer mistake, so could also panic.